client/client.go

```diff
@@ -195,14 +195,21 @@ func (c *Client) SubscribeTx(ctx context.Context, query TxQuery, results chan<-
 }
 
 // subscribe should be used internally, it wraps conn.Subscribe and uses ctx.Done() to trigger Unsubscription
-func (c *Client) subscribe(ctx context.Context, query string) (<-chan ctypes.ResultEvent, error) {
+func (c *Client) subscribe(ctx context.Context, query string, options ...Option) (<-chan ctypes.ResultEvent, error) {
+	var outCapacity []int
+	for _, option := range options {
```

Comment:
Okay, the use of some generic options seems fine, but this means we need to understand them where we use them. Do you want to make these "Subscribe Options" or some that can be used for queries (eg. maybe a TimeoutOption to auto-wrap the context??)

Reply:
I think we can just go with a single option type for now